v2/AdminCore.Services/EventService.cs

```diff
@@ -147,23 +147,49 @@ public EventDto CreateEvent(int employeeId, EventDateDto dates, EventTypes event
         EventDates = new List<EventDate>(),
         LastModified = _dateService.GetCurrentDateTime()
       };
+      if (IsHalfDay(dates))
+      {
+        newEvent.EventDates.Add(_mapper.Map<EventDate>(dates));
+      }
+      else
+      {
+        SplitEventIfFallsOnAWeekend(newEvent, dates.EndDate, dates.StartDate);
+      }
 
-      SplitEventIfFallsOnAWeekend(newEvent, dates.EndDate, dates.StartDate);
-      DatabaseContext.EventRepository.Insert(newEvent);
-      DatabaseContext.SaveChanges();
-      return _mapper.Map<EventDto>(newEvent);
+      if (DoesEmployeeHaveEnoughHolidays(employeeId, newEvent))
+      {
+        DatabaseContext.EventRepository.Insert(newEvent);
+        DatabaseContext.SaveChanges();
+        return _mapper.Map<EventDto>(newEvent);
+      }
+      throw new Exception("Not enough holidays to book");
     }
 
-    public void UpdateEvent(EventDateDto eventDateDto)
+    public void UpdateEvent(int employeeId, EventDateDto eventDateDto)
```

Comment:
Method is 22 lines long with some nesting and if ladders. Consider breaking it down into smaller chunks